### PR TITLE
Add practice exercise: killer-sudoku-helper

### PR DESCRIPTION
**Pull request by SaschaMann**
The following screenshots will be added to the examples once I figure out how to add images to exercises properly.

<details><summary>Example 1</summary>

</details>

<details><summary>Example 2</summary>

</details>

<details><summary>Example 3</summary>

</details>

### CODE REVIEW

File: exercises/practice/killer-sudoku-helper/.docs/instructions.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Instructions
+
+A friend of yours is learning how to solve Killer Sudokus but struggling to figure out which digits can go in a cage.
```
Comment (iHiD) [inline exercises/practice/killer-sudoku-helper/.docs/instructions.md:3]:
```suggestion
A friend of yours is learning how to solve Killer Sudokus but struggling to figure out which digits can go in a grid.
```

I've never heard of "cage" in this context, and [the wikipeida page](https://en.wikipedia.org/wiki/Sudoku) doesn't use the term. I've always know them (and other similar things) to be called "grids".

File: exercises/practice/killer-sudoku-helper/.docs/instructions.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+They ask you to help them out by writing a small program that lists all valid combinations for a given cage, and any constraints that affect the cage.
+
+To make the output of your program easy to read, the combinations it returns must be sorted.
+
+## Killer Sudoku Rules
+
+- [Standard Sudoku rules](https://en.wikipedia.org/wiki/Sudoku) apply.
```
Comment (iHiD) [inline exercises/practice/killer-sudoku-helper/.docs/instructions.md:10]: Not blocking, but I'll put up a one-page website with the rules of Sudoku on to make it easy to understand (without reading the whole wikipedia page, or having to go to some site that makes me accept a cookie banner!)

File: exercises/practice/killer-sudoku-helper/.docs/instructions.md (hunk 1 above)
Comment (iHiD) [reply exercises/practice/killer-sudoku-helper/.docs/instructions.md:3]: Right, I understand what a cage is now I think, but I was really confused reading this without knowing that, so I'd recommend defining "cage" before you use it. So ignore my previous suggestion, but I think we need to say in the first sentence that the rules are defined below (so someone doesn't do what I did and go googling), and then define the term "cage" in the rules.

```suggestion
A friend of yours is learning how to solve Killer Sudokus (rules below) but struggling to figure out which digits can go in a cage.
```

File: exercises/practice/killer-sudoku-helper/.docs/instructions.md (hunk 2 above)
Comment (SaschaMann) [reply exercises/practice/killer-sudoku-helper/.docs/instructions.md:10]: I could add them here too, I guess?

File: exercises/practice/killer-sudoku-helper/.docs/instructions.md (hunk 1 above)
Comment (SaschaMann) [reply exercises/practice/killer-sudoku-helper/.docs/instructions.md]: Makes sense, thanks for the improvement :)

File: exercises/practice/killer-sudoku-helper/.meta/example.jl [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+Use `collect(combinations(a, n))` to get an array of all combinations.
+"""
+function combinations(a, t::Integer)
+    if t < 0
+        # generate 0 combinations for negative argument
+        t = length(a) + 1
+    end
+    reorder(c) = [a[ci] for ci in c]
+    (reorder(c) for c in Combinations(length(a), t))
+end
```
Comment (SaschaMann) [inline exercises/practice/killer-sudoku-helper/.meta/example.jl:59]: Implementing this is kind of the main point of the exercise. However, this is only an example, not an exemplar, so I figured using a library is good enough 😅

File: exercises/practice/killer-sudoku-helper/.docs/instructions.md (hunk 2 above)
Comment (BethanyG) [reply exercises/practice/killer-sudoku-helper/.docs/instructions.md:10]: From an not-sudoku player, the pictures definitely help.  😄   I agree with @iHiD tho -- a one-pager with a clear ruleset might help as well.  The wiki article is clear....but there is a lot of detail there.

Comment (BethanyG) [reply exercises/practice/killer-sudoku-helper/.docs/instructions.md:10]: ...and having a reference/explanation of the differences might also help:  https://en.wikipedia.org/wiki/Killer_sudoku.

Comment (SaschaMann) [reply exercises/practice/killer-sudoku-helper/.docs/instructions.md:10]: I don't like linking to Wikipedia pages that start with a big "This article has multiple issues" warning but I'll try to find a resource/tutorial on killer sudokus.

Comment (BethanyG) [reply exercises/practice/killer-sudoku-helper/.docs/instructions.md:10]: Yeah.  I was going to comment on that warning at the top...but I figured that there was general "research" for directions needed.  😛

Comment (SaschaMann) [reply exercises/practice/killer-sudoku-helper/.docs/instructions.md:10]:
```suggestion
- [Standard Sudoku rules](https://masteringsudoku.com/sudoku-rules-beginners/) apply.
```

File: exercises/practice/killer-sudoku-helper/.docs/instructions.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+They ask you to help them out by writing a small program that lists all valid combinations for a given cage, and any constraints that affect the cage.
+
+To make the output of your program easy to read, the combinations it returns must be sorted.
+
+## Killer Sudoku Rules
+
+- [Standard Sudoku rules](https://masteringsudoku.com/sudoku-rules-beginners/) apply.
+- The digits in a cage, usually marked by a dotted line, add up to the small number given in the corner of the cage.
+- A digit may only occur once in a cage.
+
```
Comment (SaschaMann) [inline exercises/practice/killer-sudoku-helper/.docs/instructions.md:13]:
```suggestion
## Killer Sudoku Rules[^1]

- [Standard Sudoku rules](https://masteringsudoku.com/sudoku-rules-beginners/) apply.
- The digits in a cage, usually marked by a dotted line, add up to the small number given in the corner of the cage.
- A digit may only occur once in a cage.

[^1]: For a more detailed explanation, check out [this guide](https://masteringsudoku.com/killer-sudoku/).
```

**Pull request merged**